ti/instrumentation/instrumentation.go

```diff
@@ -30,15 +30,32 @@ func getTestSelection(ctx context.Context, runner TestRunner, config *api.RunTes
 	selection := ti.SelectTestsResp{}
 	if isManual {
 		// Manual run
-		log.Infoln("Detected manual execution - for test intelligence to be configured, a PR must be raised. Running all the tests.")
+		log.Infoln("Detected manual execution - for test intelligence to be configured the execution should be via a PR or Push trigger. Running all the tests.")
 		config.RunOnlySelectedTests = false // run all the tests if it is a manual execution
 	} else {
 		// PR execution
-		files, err := getChangedFiles(ctx, workspace, log)
+		var files []ti.File
+		var err error
+		if IsPushTriggerExecution(tiConfig) {
+			log.Infoln("Detected Push Trigger execution. Invoking Test Intelligence")
+			lastSuccessfulCommitID := getCommitInfo(ctx, stepID, tiConfig)
+			if lastSuccessfulCommitID != "" {
+				log.Infoln("Using reference commit: ", lastSuccessfulCommitID)
+			}
+			if lastSuccessfulCommitID == "" {
+				log.Infoln("Test Intelligence determined to run all the tests to bootstrap")
+				config.RunOnlySelectedTests = false // TI selected all the tests to be run
+				return selection
+			}
+			files, err = getChangedFiles(ctx, workspace, lastSuccessfulCommitID, true, log)
+		} else {
+			files, err = getChangedFiles(ctx, workspace, "", false, log)
+		}
 		if err != nil || len(files) == 0 {
-			log.Errorln("Unable to get changed files list")
+			log.Errorln("Unable to get changed files list. Running all the tests.")
 			config.RunOnlySelectedTests = false
 		} else {
+			log.Infoln("Changed Files are: ", files)
 			// PR execution: Call TI svc only when there is a chance of running selected tests
 			filesWithPkg := runner.ReadPackages(workspace, files)
 			selection, err = selectTests(ctx, workspace, filesWithPkg, config.RunOnlySelectedTests, stepID, fs, tiConfig)
```

Comment:
Make sure both source and target branch are populated in case of Push triggers

Reply:
Yeah done already by check if it is not a Manual Execution